depthmapX/mainwindowplugin.h

```diff
@@ -0,0 +1,35 @@
+// Copyright (C) 2020, Petros Koutsolampros
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#pragma once
+
+#include <QAction>
+#include <QMenuBar>
+#include <iostream>
+
+class MainWindow;
+
+class MainWindowPlugin : public QObject {
```

Comment:
Stylistic point - it would be cleaner to have an interface (i.e. only pure virtual funcs + virtual destructor), and then have an abstract implementation adding the helper methods - that way anyone *using* a plug-in would just use the interface, and only implementers would use the abstract base class.
In this case, the helpers could actually be free-standing functions - they don't use any private members of the class, so don't need to be in the class.